overlord/hookstate/hookmgr.go

```diff
@@ -143,13 +153,38 @@ func (m *HookManager) doRunHook(task *state.Task, tomb *tomb.Tomb) error {
 		return err
 	}
 
-	// TODO: Actually dispatch the hook.
+	_, err = RunHook(setup.Snap, setup.Revision, setup.Hook)
+	if err != nil {
+		exitErr, ok := err.(*exec.ExitError)
+		if ok {
+			// "exit code 1" is not particularly helpful. So rewrite the error
+			// using stderr instead.
+			err = fmt.Errorf("%s", strings.Trim(string(exitErr.Stderr), " \n"))
```

Comment:
From the docs: "Output runs the command and returns its standard output. Any returned error will usually be of type *ExitError. If c.Stderr was nil, Output populates ExitError.Stderr." And the struct for `ExitError`: https://golang.org/src/os/exec/exec.go?s=9496:10012#L342 . Note that Stderr is defined. I've tested this, it works. However, the test failure on Travis seems to be agreeing with you, which is super confusing :stuck_out_tongue: . Where did I go wrong?